torch/nn/attention/__init__.py

```diff
@@ -69,10 +69,13 @@ def sdpa_kernel(backends: List[SDPBackend]):
     This context manager can be used to select which backend to use for scaled dot product attention.
     Upon exiting the context manager, the previous state of the flags will be restored, enabling all backends.
     """
-    assert backends is None or isinstance(
```

Comment:
There were two possible fixes to this issue, thank you for making the PR. 

I think that we wanna go with the other one, where we remove the optional check and make sure that the input is a List